src/host/inputBuffer.cpp

```diff
@@ -602,6 +602,30 @@ size_t InputBuffer::Write(const std::span<const INPUT_RECORD>& inEvents)
     }
 }
 
+void InputBuffer::WriteDirect(const std::span<const INPUT_RECORD>& inEvents)

```

Comment:
Instead of creating a new method for this, couldn't you just make the existing `_HandleTerminalInputCallback` method public, and call it with a string instead of an event list? That way you could drop all that event generation code from the `ReturnResponse` method, and pass the string through directly.

Reply:
Yeah, that's a good point. I'll make that change.